src/movement_controls.py

```diff
@@ -0,0 +1,316 @@
+    acceleration_rate: float
+    # how fast the brake applies to current acceleration
+    deceleration_rate: float
+    # how fast the vehicle's current acceleration slows while not accelerating
+    friction: float
+
+    # These are the values the vehicle updates in "drive_input"
+    current_velocity_x: float
+    current_velocity_y: float
+    current_turnVelocity: float
+    current_acceleration: float
+
+    # which control scheme to use
+    vehicle_type: int
+    # for switching between vehicle types
+    toggle: bool
+
+    def __init__(self):
+        self.current_velocity_x = 0
+        self.current_velocity_y = 0
+        self.current_turnVelocity = 0
+        self.current_acceleration = 0
+        self.acceleration_rate = 0.5
+        self.decceleration_rate = 0.3
+        self.brake_rate = 0.3
+        self.friction = 0.1
+        self.drive_speed = 100
+        self.turn_speed = 100
+        self.toggle = False
+        self.vehicle_type = 1
+
+    #   Drive input is called by the player and passed that player's input to set what the vehicle's
+    #    velocity and rotation should be
+    #   Does not change the vehicles position
+    def drive_input(self, delta_time, input, vehicle):
+        # for testing to change between driving types
+        # TODO:find a permanent place for this logic that is not in movement
+        if self.toggle:
+            if input.secondary_fire_button.value == False:
+                self.toggle = False
+        else:
+            if input.secondary_fire_button.value == True:
+                self.toggle = True
+                self.change_vehicle()
+
+        # updates the vehicles intended  velocity and rotation based on type
+        if self.vehicle_type == 0:
+            self.car(delta_time, input, vehicle)
+        elif self.vehicle_type == 1:
+            self.ghost(delta_time, input, vehicle)
+        elif self.vehicle_type == 2:
+            self.ghost_screen_facing(delta_time, input, vehicle)
+        elif self.vehicle_type == 3:
+            self.tank(delta_time, input, vehicle)
+        elif self.vehicle_type == 4:
+            self.mars_walker(delta_time, input, vehicle)
+
+    def change_vehicle(self):
+        self.current_acceleration = 0
+        self.current_velocity_x = 0
+        self.current_velocity_y = 0
+        self.current_turnVelocity = 0
+        self.vehicle_type += 1
+        if self.vehicle_type > 4:
+            self.vehicle_type = 0
+
+    # called from the player to tell the vehicle to act on it's intended velocity and rotation
+    def move(self, vehicle):
+        # find the X and Y movement based on the angle of the sprite
+        new_x_pos = self.current_velocity_x + vehicle.center_x
+        new_y_pos = self.current_velocity_y + vehicle.center_y
+
+        # NOTE: place holder boundaries
+        if new_x_pos < 0 or new_x_pos > 500 or new_y_pos < 0 or new_y_pos > 500:
+            # bounce of the imaginary collision and roll away
```

Comment:
Could you extend these to the edge of the screen so we can debug with the whole area. You can use SCREEN_WIDTH and
SCREEN_HEIGHT from constants.py